gradio/components.py

```diff
@@ -3937,8 +3939,296 @@ def postprocess(self, y: str | None) -> Dict[str, str] | None:
             out_y = y.to_json()
         return {"type": dtype, "plot": out_y}
 
-    def style(self):
-        return self
+    def style(self, container: Optional[bool] = None):
+        return IOComponent.style(
+            self,
+            container=container,
+        )
+
+
+@document("change", "clear")
+class ScatterPlot(Plot):
+    """
+    Create a scatter plot.
+
+    Preprocessing: this component does *not* accept input.
+    Postprocessing: expects a pandas dataframe with the data to plot.
+
+    Demos: native_plots
+    """
+
+    def __init__(
+        self,
+        x: Optional[str] = None,
+        y: Optional[str] = None,
+        value: Optional[pd.DataFrame] = None,
+        color: Optional[str] = None,
+        size: Optional[str] = None,
+        shape: Optional[str] = None,
+        title: Optional[str] = None,
+        tooltip: Optional[List[str] | str] = None,
+        x_title: Optional[str] = None,
+        y_title: Optional[str] = None,
+        color_legend_title: Optional[str] = None,
+        size_legend_title: Optional[str] = None,
+        shape_legend_title: Optional[str] = None,
+        height: Optional[int] = None,
+        width: Optional[int] = None,
+        caption: Optional[str] = None,
+        interactive: Optional[bool] = True,
```

Comment:
Should we use a different parameter than `interactive`? For all of the other components, `interactive` means whether or not the component is used as an input or an output -- if we were to extend plots to be input components like we've considered, it might be helpful to have a separate parameter

Reply:
Never mind, I think `interactive` might actually be appropriate here. We don't need to overthink it